database_admin/migrations/124_templates.up.sql

```diff
@@ -0,0 +1,50 @@
+CREATE OR REPLACE FUNCTION grant_table_partitions(perms text, tbl regclass, grantie text)
+    RETURNS VOID AS
+$$
+DECLARE
+    r record;
+BEGIN
+    FOR r IN SELECT child.relname
+               FROM pg_inherits
+               JOIN pg_class parent
+                 ON pg_inherits.inhparent = parent.oid
+               JOIN pg_class child
+                 ON pg_inherits.inhrelid   = child.oid
+              WHERE parent.relname = text(tbl)
+    LOOP
+        EXECUTE 'GRANT ' || perms || ' ON TABLE ' || r.relname || ' TO ' || grantie;
+    END LOOP;
+    EXECUTE 'GRANT ' || perms || ' ON TABLE ' || text(tbl) || ' TO ' || grantie;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TABLE IF NOT EXISTS template
+(
+    id            BIGINT            GENERATED BY DEFAULT AS IDENTITY,
+    rh_account_id INT               NOT NULL REFERENCES rh_account (id),
+    uuid          UUID              NOT NULL,
+    name          TEXT              NOT NULL CHECK (not empty(name)),
+    description   TEXT              CHECK (NOT empty(description)),
+    config        JSONB,
+    creator       TEXT              CHECK (NOT empty(creator)),
```

Comment:
Maybe we can actually remove it because it is not in in a current template message. But I'm afraid customers will want it (they have it in baselines).

Reply:
okay, let's keep it in DB schema in case content-sources needs to add info about creator.